flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -352,6 +354,24 @@ static mlir::Value genDefaultInitializerValue(
   return initialValue;
 }
 
+/// Does this global already have an initializer ?
+static bool globalIsInitialized(fir::GlobalOp global) {
+  return !global.getRegion().empty() || global.initVal();
+}
+
+/// Call \p genInit to generate code inside \p global initializer region.
+static void
+createInitialRegion(fir::FirOpBuilder &builder, fir::GlobalOp global,
```

Comment:
I think this means "create the initializer region" rather than "create the initial region".

Maybe rename this to `createGlobalInitialization`.

(A global need not have any region. If it is a declaration it must not have one, and even a definition can omit the region and be initialized with an Attribute.)